elements/lisk-codec/src/json_wrapper.ts

```diff
@@ -0,0 +1,180 @@
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import {
+	BaseTypes,
+	GenericObject,
+	SchemaPair,
+	SchemaProps,
+	SchemaScalarItem,
+} from './types';
+
+export interface IteratableGenericObject extends GenericObject {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	[Symbol.iterator](): Iterator<{ key: string; value: any }>;
+}
+
+const _liskMessageValueToJSONValue: {
+	readonly [key: string]: (value: BaseTypes) => BaseTypes;
+} = {
+	uint32: value => value as number,
+	sint32: value => value as number,
+	uint64: value => (value as BigInt).toString(),
+	sint64: value => (value as BigInt).toString(),
+	string: value => value as string,
+	bytes: value => (value as Buffer).toString('base64'),
+	boolean: value => value as boolean,
+};
+
+const _JSONValueToliskMessageValue: {
+	readonly [key: string]: (value: BaseTypes) => BaseTypes;
+} = {
+	uint32: value => value as number,
+	sint32: value => value as number,
+	uint64: value => BigInt(value),
+	sint64: value => BigInt(value),
+	string: value => value as string,
+	bytes: value => Buffer.from(value as string, 'base64'),
+	boolean: value => value as boolean,
+};
+
+const findObjectByPath = (
+	message: SchemaProps,
+	pathArr: string[],
+): SchemaProps | undefined => {
+	let result: SchemaProps = message;
+	for (let i = 0; i < pathArr.length; i += 1) {
+		if (!result.properties && !result.items) {
+			return undefined;
+		}
+		if (result.properties) {
+			result = result.properties[pathArr[i]];
+		} else if (result.items) {
+			const x = (result.items as SchemaProps).properties as SchemaPair;
+			result = x[pathArr[i]];
+		}
+	}
+	return result;
+};
+
+const isObject = (item: unknown): boolean =>
+	typeof item === 'object' &&
+	item !== null &&
+	!Array.isArray(item) &&
+	!Buffer.isBuffer(item);
+
+export const iterator = function iterator(
+	this: IteratableGenericObject,
+): {
+	next: () => {
+		done: boolean;
+		value: { value: IteratableGenericObject; key: string };
+	};
+} {
+	let index = 0;
+	const properties = Object.keys(this);
+	let Done = false;
+	return {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		next: (): { done: boolean; value: { value: any; key: string } } => {
+			Done = index >= properties.length;
+			const obj = {
+				done: Done,
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+				value: { value: this[properties[index]], key: properties[index] },
+			};
+			index += 1;
+			return obj;
+		},
+	};
+};
+
+export const recursiveTypeCast = (
+	mode: 'toJSON' | 'fromJSON',
+	object: IteratableGenericObject,
+	schema: SchemaProps,
+	dataPath: string[],
+): void => {
+	for (const { key, value } of object) {
+		if (isObject(value)) {
```

Comment:
Instead of this native check can' we use schema defined `type: object`

Reply:
This is a check to go into recursion if value is an object literal; I think the TS type will not work for this?

Reply:
OK got clarification about this. I prefer to keep the check by observing the object itself.

Reply:
No not TS type. If its an object, then relevant schema definition must have `type: object` for it. So why not use that?